lib/pbench/server/api/resources/datasets_inventory.py

```diff
@@ -53,37 +54,43 @@ def _get(
         Raises:
             APIAbort, reporting either "NOT_FOUND" or "UNSUPPORTED_MEDIA_TYPE"
 
-
         GET /api/v1/datasets/{dataset}/inventory/{target}
         """
         dataset = params.uri["dataset"]
         target = params.uri.get("target")
 
         cache_m = CacheManager(self.config, current_app.logger)
         try:
-            file_info = cache_m.filestream(dataset, target)
+            file_info = cache_m.filestream(dataset.resource_id, target)
         except TarballNotFound as e:
             raise APIAbort(HTTPStatus.NOT_FOUND, str(e))
 
-        if file_info["type"] == CacheType.FILE:
-            # Tell send_file to set `Content-Disposition` to "attachment" if
-            # targeting the large binary tarball. Otherwise we'll recommend the
-            # default "inline": only `is_file()` paths are allowed here, and
-            # most are likely "displayable". While `send_file` will guess the
-            # download_name from the path, setting it explicitly does no harm
-            # and supports a unit test mock with no real file.
-            #
-            # NOTE: we could choose to be "smarter" based on file size, file
-            # type codes (e.g., .csv, .json), and so forth.
-            resp = send_file(
-                file_info["stream"],
-                as_attachment=target is None,
-                download_name=file_info["name"],
-            )
-            file_info["stream"].close()
-            return resp
-        else:
+        if file_info["type"] != CacheType.FILE:
             raise APIAbort(
                 HTTPStatus.BAD_REQUEST,
                 "The specified path does not refer to a regular file",
             )
```

Comment:
Not here: that would close it before it's actually used. That's what the `call_on_close` is for at line 95.

Reply:
`filestream` only returns a stream for `CacheType.FILE`, so there won't be one in the `else` clause. However I could conditionally close it here if it were to ever come to exist in the future, at the expense of an extra `if` that's currently a complete waste.

Reply:
Ah, right...nevermind.

(Although, `tarfile.extractfile()` also returns a stream if the target is a symlink, which we'll report as a `CacheType.FILE`, which is mostly fine, but the [documentation](https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.extractfile) doesn't indicate whether the stream contents are the symlink itself or that of the referent...but, I don't really see how it would make sense (i.e., in the general case) for it to be the referent, so I assume the stream contains the contents of the symlink file.)